### PR TITLE
ENYO-2145: More robust handling of values.

### PR DESCRIPTION
**Pull request by aarontam**
### Issue

The LESS API is fairly lacking and the initial pass of the resolution-independence plugin felt more like fumbling around in the dark. Now that we have a better idea of the API's behavior, we can take a step back and rework the parsing logic to more broadly handle different cases we will encounter.
### Fix

The parsing has been reworked to handle a variety of cases by recursively parsing each "value" until it can be resolved via conversion or is a string that does not require conversion. Additionally, a set of tests has been added to ensure we properly parse and handle all of these cases. These can be run via `npm test` in the root directory.

Enyo-DCO-1.1-Signed-off-by: Aaron Tam aaron.tam@lge.com

### CODE REVIEW

File: test/resolution-independence.js [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+	Less = require('../node_modules/less/lib/less');
+
+describe('resolution-independence options', function () {
+	var RezInd, opts, riPlugin, node;
+
+	before(function () {
+		// TODO: we initialize the plugin with a dummy LESS object for now, so that we can test
+		// specific methods of the plugin (this is done using a mock LESS node) - ideally we will
+		// test with the actual nodes generated by LESS, in case the format changes, but this will
+		// require the likely creation of a plugin to hook into the returned nodes from LESS, in
```
Comment (ryanjduffy) [inline test/resolution-independence.js:13]: yo dawg, i heard you like plugins so i built a plugin so you could test your plugin.

**Comment by ryanjduffy** [comment]
We should update the version in package.json too, right?

**Comment by aarontam** [comment]
No! I mean, yes... :wink:

File: test/resolution-independence.js (hunk 1 above)
Comment (aarontam) [inline test/resolution-independence.js:13]: I meant to add a (yodawg) to the comments but forgot (facepalm)

**Pull request merged**